joomla/joomla-cms
# Microdata implementation in the CMS with the use of plg_system_microdata and JMicrodata

This pull request includes the `plg_system_microdata` plugin https://github.com/joomla/joomla-cms/pull/4118 pull request.
## THINGS DISCUSSED

1) Where is saved the com_content microdata information.
Discussion link: https://twitter.com/PAlexcom/status/513414081534263297
![microdata-database](https://cloud.githubusercontent.com/assets/1648497/4346043/c45d1664-40ec-11e4-8485-2c87d4f5118a.png)
## HOW TO TEST

First I recommend reading this article http://magazine.joomla.org/issues/issue-aug-2014/item/2233-rich-snippets-structured-data-microdata-rdfa-isnt-so-simple

Also make sure you have the `plg_content_vote` and `plg_system_microdata` both enabled.

You will also need this tool provided by google to test the markup and the SERP (Search Engine Result Page) preview https://www.google.com/webmasters/tools/richsnippets

Not necessary, but you could also validate the HTML by using this tool http://validator.w3.org
## Step by step
1. Go in the Administration area, then from the top menu select: _Content → Article Manager_
2. Now select and press _Edit_ on any article
3. From the option tab select: _Publishing_ ![administration - article manager edit article](https://cloud.githubusercontent.com/assets/1648497/4260976/317a139a-3b58-11e4-9a68-28f3ebe98eab.png)
4. At the bottom right of the page you should find: ![administration - article manager edit article microdata](https://cloud.githubusercontent.com/assets/1648497/4260980/615083a6-3b58-11e4-9c8b-c83f4ead2c5f.png)
## Previews and results

At this point you can change one of the 5 available microdata schema.org types.
### Article
##### Backend configuration:

![admin - administration - article manager edit article microdata-article](https://cloud.githubusercontent.com/assets/1648497/4261013/b5deaa5a-3b59-11e4-8797-45b328ebed42.png)
##### Preview of the rich snippet in the SERP of Google:

![google structured data testing tool article](https://cloud.githubusercontent.com/assets/1648497/4261016/ce2446f6-3b59-11e4-8c55-4fdb60eb6074.png)
### Event
##### Backend configuration:

![administration - article manager edit article event](https://cloud.githubusercontent.com/assets/1648497/4261026/6c9dc67c-3b5a-11e4-8b5b-30634b5abec9.png)
##### Frontend output:

![beginners frontend event](https://cloud.githubusercontent.com/assets/1648497/4261029/7f167fba-3b5a-11e4-99ce-184ff2b69152.png)
##### Preview of the rich snippet in the SERP of Google:

![google structured data testing tool event](https://cloud.githubusercontent.com/assets/1648497/4261028/75785c44-3b5a-11e4-851e-887c0e5f55fa.png)
### Product
##### Backend configuration:

![administration - article manager edit article product](https://cloud.githubusercontent.com/assets/1648497/4261034/be5d1e54-3b5a-11e4-88bc-2ef8636a0ff4.png)
##### Preview of the rich snippet in the SERP of Google:

![google structured data testing tool product](https://cloud.githubusercontent.com/assets/1648497/4261035/c0bc8392-3b5a-11e4-8975-5c189cc887de.png)
### Recipe
##### Backend configuration:

![administration - article manager edit article recipe](https://cloud.githubusercontent.com/assets/1648497/4261041/fa4dbc20-3b5a-11e4-8283-677eaf369657.png)
##### Frontend output:

![beginners frontend recipe](https://cloud.githubusercontent.com/assets/1648497/4261043/01ebe1e6-3b5b-11e4-9a3f-c3e315422540.png)
##### Preview of the rich snippet in the SERP of Google:

![google structured data testing tool recipe](https://cloud.githubusercontent.com/assets/1648497/4261047/0d37d262-3b5b-11e4-808d-819eba7f1160.png)
### Review
##### Backend configuration:

![administration - article manager edit article review](https://cloud.githubusercontent.com/assets/1648497/4261051/1ecd46d8-3b5b-11e4-8b81-a43ccaad0794.png)
##### Preview of the rich snippet in the SERP of Google:

![google structured data testing tool review](https://cloud.githubusercontent.com/assets/1648497/4261052/262db8e0-3b5b-11e4-8437-8d449537e09d.png)

---

For more information on how everything is implemented and the syntax please read here https://github.com/joomla/joomla-cms/pull/4118
